data_cleaning/ice_velocity.ipynb

```diff
@@ -0,0 +1,468 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "667e10c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var_ls = list(ds)\n",
+    "\n",
+    "yr_ls = list(set([var[-4:] for var in var_ls]))\n",
+    "\n",
+    "yr_ls = sorted([pd.to_datetime(year).year for year in yr_ls])\n",
+    "yr_ls"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "26618522-cdb9-4ec9-b267-a96b44f96e21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vx_ls = [var for var in var_ls if 'vx' in var]\n",
+    "vy_ls = [var for var in var_ls if 'vy' in var]\n",
+    "err_ls = [var for var in var_ls if 'err' in var]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29a73217",
+   "metadata": {},
+   "source": [
+    "Now we are going to group the `dataset.data_vars` into `vx`,`vy`, and `err` and prepare to concatenate them along the time dimension. We will perform the same operations for all three variables but we will demonstrate the process for the first variable in several steps, before showing the operation wrapped into one command for the other two variables. There is a great explanation of this kind of step [here](https://towardsdatascience.com/pythonic-way-to-perform-statistics-across-multiple-variables-with-xarray-d0221c78e34a). At the end of this step, for `vx`, `vy` and `err` we will have a list of `xr.DataArrays` that all have a time dimension on the 0-axis. \n",
+    "\n",
+    "In the cell below, we make a list of the `xr.DataArrays` in the original `xr.Dataset` that correspond to that variable. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9360bc75",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da_vx_ls = [ds[var] for var in vx_ls]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0ed3f37e",
+   "metadata": {},
+   "source": [
+    "You can see that `da_vx_ls` is a `list` object with a length of 10, and each element of the list is a `xr.DataArray` corresponding to `vx` vars in the original `xr.Dataset`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f81b86ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('Object type: ', type(da_vx_ls))\n",
+    "print('Ojbect length: ', len(da_vx_ls))\n",
+    "da_vx_ls[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66231382",
+   "metadata": {},
+   "source": [
+    "next, we will add a time dimension to every element of `da_vx_ls`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e84a673f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da_vx_ls = [da_vx_ls[var].expand_dims('time', axis=0) for var in range(len(da_vx_ls))]"
```

Comment:
can simplify this by iterating on the dataarrays themselves: `da_vx_ls = [da_vx.expand_dims('time', axis=0) for da_vx in da_vx_ls]`